tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Couchbase/CouchbaseCommon.cs

```diff
@@ -124,5 +128,70 @@ internal static TOperationResult CommonOnMethodEnd<TOperationResult>(TOperationR
             state.Scope.DisposeWithException(exception ?? result.Exception);
             return tResult;
         }
+
+        internal static string GetNormalizedSeedNodesFromClientConfiguration(IClientConfiguration clientConfiguration)
+        {
+            if (ClientSourceToNormalizedSeedNodesMap.TryGetValue(clientConfiguration.Instance, out var normalizedSeedNodes))
+            {
+                return normalizedSeedNodes;
+            }
+
+            // Construct the normalized value from the list of hosts (each host will be {Host} or {Host}:{Port})
+            var sb = StringBuilderCache.Acquire(0);
+            IList<Uri> servers = clientConfiguration.Servers;
+
+            for (int i = 0; i < servers.Count; i++)
+            {
+                if (i != 0)
+                {
+                    sb.Append(',');
+                }
+
+                var uri = servers[i];
+                sb.Append($"{uri.Host}:{uri.Port}");
+            }
+
+            normalizedSeedNodes = StringBuilderCache.GetStringAndRelease(sb);
+
+#if NETCOREAPP3_1_OR_GREATER
```

Comment:
We start doing this dance in so many places, I wonder if it wouldn't make sense to have our own `ConditionalWeakTable` override that would just hide the frameworks differences for us. But just seen that - Oh sadness - `ConditionalWeakTable` is sealed so :homerdisappear: